contracts/FixedPriceOracle.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.4.24;
+
+/**
+ * @title An Ether-ERC20 token price oracle with an unmutable price
+ * 
+ * @dev The prices are initialized when configuring the contract, then you 
+ *  freeze the contract disallowing any further modification.
+ */ 
+contract FixedPriceOracle {
+    mapping (address => Price) public prices;
+    bool public frozen;
+    address public owner;
+    
+    modifier onlyOwner () {
+        require(msg.sender == owner, "Only the owner can do the operation");
+        _;
+    }
+    
+    modifier notFrozen () {
+        require(!frozen, "The contract is frozen, not changes are allowed");
+        _;
+    }
+    
+    struct Price {
+        uint numerator;
+        uint denominator;
+    }
+    
+    event PriceSet(
+         address indexed token,
+         uint numerator,
+         uint denominator
+    );
+    
+    event Freeze();
+    
+    
+    constructor () public {
+        owner = msg.sender;
+    }
+
+    function hasReliablePrice (address token) public view returns (bool) {
+        return prices[token].denominator != 0;
+    }
+    
+    function getPrice(address token) public view returns (uint, uint) {
+        Price memory price = prices[token];
+        return (price.numerator, price.denominator);
+    }
+    
+    function setPrices(
+        address[] tokens,
+        uint[] numerators,
+        uint[] denominators
+    ) public onlyOwner notFrozen {
+        for (uint i = 1 ; i < tokens.length ; i++) {
+            address token = tokens[i];
+            uint numerator = numerators[i];
```

Comment:
Any reason you don't do `setPrice(tokens[i], numerators[i], denominators[i]);` here? Is it more gas-efficient to inline?

Reply:
Yes, so I don't have to access the arrays 2 times. Notice the emit of the event after that.